vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/devtools/MobileEmulationSteps.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.steps.ui.web.devtools;
+
+import static org.apache.commons.lang3.Validate.isTrue;
+
+import java.util.Map;
+
+import org.jbehave.core.annotations.When;
+import org.openqa.selenium.chromium.HasCdp;
+import org.openqa.selenium.remote.Browser;
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.selenium.manager.IWebDriverManager;
+import org.vividus.util.json.JsonUtils;
+
+public class MobileEmulationSteps
+{
+    private final IWebDriverProvider webDriverProvider;
+    private final IWebDriverManager webDriverManager;
+    private final JsonUtils jsonUtils;
+
+    public MobileEmulationSteps(IWebDriverProvider webDriverProvider, IWebDriverManager webDriverManager,
+            JsonUtils jsonUtils)
+    {
+        this.webDriverProvider = webDriverProvider;
+        this.webDriverManager = webDriverManager;
+        this.jsonUtils = jsonUtils;
+    }
+
+    /**
+     * Emulates mobile device using provided configuration.
+     *
+     * <p>
+     * <strong>The step is only supported in Chrome browser.</strong>
+     * </p>
+     *
+     * @param jsonConfiguration The JSON containing device metrics to override.
+     */
+    @SuppressWarnings("unchecked")
+    @When("I emulate mobile device with configuration:`$jsonConfiguration`")
+    public void overrideDeviceMetrics(String jsonConfiguration)
+    {
+        executeCdpCommand("Emulation.setDeviceMetricsOverride", jsonUtils.toObject(jsonConfiguration, Map.class));
+    }
+
+    /**
+     * Resets the mobile device emulation returning browser screen to its initial state.
+     *
+     * <p>
+     * <strong>The step is only supported in Chrome browser.</strong>
+     * </p>
+     */
+    @When("I reset mobile device emulation")
+    public void clearDeviceMetrics()
+    {
+        executeCdpCommand("Emulation.clearDeviceMetricsOverride", Map.of());
```

Comment:
what if the browser was started in mobile emulation mode?

Reply:
it will reset the mobile emulation, even if we have mobile emulation profile activated, technically the initial state of the browser doesn't include mobile emulation, its turned on after the browser is loaded